scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -424,6 +444,9 @@ def __init__(self, configuration, monitors):
             self.__config, self.__monitors
         )  # type: List[LogMatcher]
 
+        # this dict stores all checkpoint states for all current and closed files
+        self._checkpoints = {}
```

Comment:
I believe this means our memory usage will grow every time if we start copying a new file.  In the case of K8s, where pods keep coming up and down, this list will grow unbounded.

I would like us to try to make sure our memory usage is only proportional to the number of files we are currently copying.  Previously, we just made sure we read from disk all checkpoints (including files we are no longer processing) when we were looking for old checkpoints.  So, there was only a limited time that we held onto that memory.  I'd like you to think a little more if there's some way we can avoid holding onto all of these checkpoints all of the time.